lab_stack/ConsoleApplication33/ConsoleApplication33/Stack.cpp

```diff
@@ -57,69 +56,69 @@
         }
     }
 
-    //�������� �����
+    //î÷èùåíèå ñòåêà
     void Stack::Clear()
     {
         Stack::~Stack();

```

Comment:
The same. You dont need destructor here. first element will be nullptr

Reply:
But we lose access to the next elements, not right? And it means, that we can't free memory these elements. Or am I misunderstanding?

Reply:
NEVER call a destructor explicitly. It will be Undefined Behavior. (Destuctor will be called twice. First - your call; Second - after lifecycle)
An exception - placement new
But u dont need to understand now what is it UB and placement new. 
Remember - dont call a destructor explicitly.
Compiler will call destructor automaticaly.
 Think carefully about what happens when you clear the stack.